modules/nf-core/jvarkit/vcf2table/main.nf

```diff
@@ -0,0 +1,63 @@
+process JVARKIT_VCF2TABLE {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/jvarkit:2024.08.25--hdfd78af_1':
+        'biocontainers/jvarkit:2024.08.25--hdfd78af_1' }"
+
+    input:
+    tuple val(meta), path(vcf), path(tbi), path(regions_file)
+    tuple val(meta2), path(pedigree)
+    output:
+    tuple val(meta), path("*.${extension}"), emit: output
+    path "versions.yml"                    , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args1        = task.ext.args1 ?: ''
+    def args2        = task.ext.args2 ?: ''
+    def prefix       = task.ext.prefix ?: "${meta.id}"
+    def ped          = pedigree?"--pedigree \"${pedigree}\"":""
+    def regions_file = regions_file? (tbi ? " --regions-file" : " --targets-file")+" \"${regions_file}\" ":""
```

Comment:
Why does this change between being regions or targets based on the presence of the index?

Reply:
bcftools needs an index for the option **--regions-file** . But when there is no index, `--targets-file` must be used, it scans the **whole** file and keeps the variants in the regions.

see then bcftools man:

https://samtools.github.io/bcftools/bcftools.html

```
-T, --targets-file [^]FILE

    Same -t, --targets, but reads regions from a file.
-t, --targets [^]chr|chr:pos|chr:from-to|chr:from-[,…​]
    Similar as -r, --regions, but the next position is accessed by streaming the whole VCF/BCF rather than using the tbi/csi index.


```